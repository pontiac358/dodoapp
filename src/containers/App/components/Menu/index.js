import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom'
import styled from '@emotion/styled'

const Container = styled.div``;

const MenuWrap = styled.div`
    @media only screen and (max-width: 480px) {
        position: absolute;
        top: 48px;
        left:0;
        background:#000;
        width: 100%;
        display:${props => props.isOpen ? 'block' : 'none'};
        padding: 10px;
        text-align:center;   
    }
`;

const MobileMenuIcon = styled.div`
    display:none;
    @media only screen and (max-width: 480px) {
        display:block;   
    }
`;


const MenuLink = styled(Link)`
  padding: 10px;
  color: #fff;
  @media only screen and (max-width: 480px) {
        display:block;   
    }
`;
MenuLink.displayName = 'MenuLink';


class Menu extends React.Component  {
    state = {
        isOpen: false
    };

    menuToggle = () => {
        this.setState({isOpen: !this.state.isOpen})
    };

    render() {
        const { menu } = this.props;

        return (
            <Container>
                <MobileMenuIcon onClick={this.menuToggle}>
                    Menu
                </MobileMenuIcon>
                <MenuWrap isOpen={ this.state.isOpen }>
                    {
                        menu.data && menu.data.map(item => <MenuLink key={ item.id } to={`/menu/${item.id}/${item.url}`}> { item.title } </MenuLink>)
                    }
                </MenuWrap>
            </Container>
        );
    }

}

Menu.propTypes = {
    menu: PropTypes.shape({
        data: PropTypes.array,
        isLoading: PropTypes.bool
    }),
}

export default Menu;