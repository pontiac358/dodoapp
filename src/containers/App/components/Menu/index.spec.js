import React from 'react'
import { shallow } from 'enzyme'
import Menu from '../Menu'

describe('Menu component', () => {
    const props = {
        menu:{
            data: [
                {
                    title: 'test',
                    url: 'test',
                    id: 1
                }
            ],
            isLoading: false,
        }
    };

    const wrapper = shallow(<Menu {...props} />);

    it("render correctly ProductList container", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('renders <MenuLink /> wrapper', () => {
        expect(wrapper.find('MenuLink')).toHaveLength(props.menu.data.length)
    })

});

