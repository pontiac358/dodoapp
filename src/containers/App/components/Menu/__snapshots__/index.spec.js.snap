// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component render correctly ProductList container 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Menu
    menu={
      Object {
        "data": Array [
          Object {
            "id": 1,
            "title": "test",
            "url": "test",
          },
        ],
        "isLoading": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef(render)
          onClick={[Function]}
        >
          Menu
        </ForwardRef(render)>,
        <ForwardRef(render)
          isOpen={false}
        >
          <ForwardRef(render)
            replace={false}
            to="/menu/1/test"
          >
             
            test
             
          </ForwardRef(render)>
        </ForwardRef(render)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Menu",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Menu",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "__emotion_base": "div",
          "__emotion_forwardProp": undefined,
          "__emotion_real": [Circular],
          "__emotion_styles": Array [
            "
    display:none;
    @media only screen and (max-width: 480px) {
        display:block;   
    }
",
          ],
          "defaultProps": undefined,
          "displayName": "Styled(div)",
          "render": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef(render)
              replace={false}
              to="/menu/1/test"
            >
               
              test
               
            </ForwardRef(render)>,
          ],
          "isOpen": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "children": Array [
                " ",
                "test",
                " ",
              ],
              "replace": false,
              "to": "/menu/1/test",
            },
            "ref": null,
            "rendered": Array [
              " ",
              "test",
              " ",
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "__emotion_base": [Function],
              "__emotion_forwardProp": undefined,
              "__emotion_real": [Circular],
              "__emotion_styles": Array [
                "
  padding: 10px;
  color: #fff;
  @media only screen and (max-width: 480px) {
        display:block;   
    }
",
              ],
              "defaultProps": Object {
                "replace": false,
              },
              "displayName": "MenuLink",
              "render": [Function],
              "withComponent": [Function],
            },
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "__emotion_base": "div",
          "__emotion_forwardProp": undefined,
          "__emotion_real": [Circular],
          "__emotion_styles": Array [
            "
    @media only screen and (max-width: 480px) {
        position: absolute;
        top: 48px;
        left:0;
        background:#000;
        width: 100%;
        display:",
            [Function],
            ";
        padding: 10px;
        text-align:center;   
    }
",
          ],
          "defaultProps": undefined,
          "displayName": "Styled(div)",
          "render": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "__emotion_base": "div",
      "__emotion_forwardProp": undefined,
      "__emotion_real": [Circular],
      "__emotion_styles": Array [
        "",
      ],
      "defaultProps": undefined,
      "displayName": "Styled(div)",
      "render": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef(render)
            onClick={[Function]}
          >
            Menu
          </ForwardRef(render)>,
          <ForwardRef(render)
            isOpen={false}
          >
            <ForwardRef(render)
              replace={false}
              to="/menu/1/test"
            >
               
              test
               
            </ForwardRef(render)>
          </ForwardRef(render)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Menu",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Menu",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "__emotion_base": "div",
            "__emotion_forwardProp": undefined,
            "__emotion_real": [Circular],
            "__emotion_styles": Array [
              "
    display:none;
    @media only screen and (max-width: 480px) {
        display:block;   
    }
",
            ],
            "defaultProps": undefined,
            "displayName": "Styled(div)",
            "render": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ForwardRef(render)
                replace={false}
                to="/menu/1/test"
              >
                 
                test
                 
              </ForwardRef(render)>,
            ],
            "isOpen": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  " ",
                  "test",
                  " ",
                ],
                "replace": false,
                "to": "/menu/1/test",
              },
              "ref": null,
              "rendered": Array [
                " ",
                "test",
                " ",
              ],
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "__emotion_base": [Function],
                "__emotion_forwardProp": undefined,
                "__emotion_real": [Circular],
                "__emotion_styles": Array [
                  "
  padding: 10px;
  color: #fff;
  @media only screen and (max-width: 480px) {
        display:block;   
    }
",
                ],
                "defaultProps": Object {
                  "replace": false,
                },
                "displayName": "MenuLink",
                "render": [Function],
                "withComponent": [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "__emotion_base": "div",
            "__emotion_forwardProp": undefined,
            "__emotion_real": [Circular],
            "__emotion_styles": Array [
              "
    @media only screen and (max-width: 480px) {
        position: absolute;
        top: 48px;
        left:0;
        background:#000;
        width: 100%;
        display:",
              [Function],
              ";
        padding: 10px;
        text-align:center;   
    }
",
            ],
            "defaultProps": undefined,
            "displayName": "Styled(div)",
            "render": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "__emotion_base": "div",
        "__emotion_forwardProp": undefined,
        "__emotion_real": [Circular],
        "__emotion_styles": Array [
          "",
        ],
        "defaultProps": undefined,
        "displayName": "Styled(div)",
        "render": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
